From b4f3f17e292eb8639a0fb7435e61ffa917d986b1 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 09 Jan 2020 16:06:00 +0200
Subject: [PATCH] [SW-6739] arcboot: enable icache and dcache

to improve hash and decryption preformace of synopsis
library we turned on instruction and data caches so
arc wont have to constantly fetch from external buses
and slow the exection of the code

Change-Id: I78ef64e3303339783feb4ddcda15b7ce06f4ed8d
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/crt0_rom.S b/common/crt0_rom.S
index 1a43094..d4680ee 100644
--- a/common/crt0_rom.S
+++ b/common/crt0_rom.S
@@ -13,6 +13,15 @@
 .file "crt0_rom.S"
 
 .macro CPU_EARLY_SETUP
+	; enable I-cache/D-cache
+	lr	r5, [ic_ctrl]
+	bclr	r5, r5, 0	; 0 - Enable, 1 is Disable
+	sr	r5, [ic_ctrl]
+
+	lr	r5, [dc_ctrl]
+	bclr	r5, r5, 6	; Invalidate (discard w/o wback)
+	bclr	r5, r5, 0	; Enable (+Inv)
+	sr	r5, [dc_ctrl]
 
 	; Disable alignment checking
 	lr	r5, [status32]
